openshift/hni-patroni-postgresql-prerequisite.yml

```diff
@@ -0,0 +1,148 @@
+      replication-password: ${PATRONI_REPLICATION_PASSWORD}
+      superuser-username: ${PATRONI_SUPERUSER_USERNAME}
+      superuser-password: ${PATRONI_SUPERUSER_PASSWORD}
+      app-db-name: ${APP_DB_NAME}
+      app-db-username: app_api_${APP_DB_USERNAME}
+      app-db-password: ${APP_DB_PASSWORD}
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      labels:
+        app: ${APP_LABEL_NAME}
+        cluster-name: ${NAME}
+      name: ${NAME}
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    metadata:
+      labels:
+        app: ${APP_LABEL_NAME}
+        cluster-name: ${NAME}
+      name: ${NAME}
+    rules:
+      - apiGroups:
+          - ""
+        resources:
+          - services
+        verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+          - delete
+      - apiGroups:
+          - ""
+        resources:
+          - configmaps
+        verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+          - delete
+      - apiGroups:
+          - ""
+        resources:
+          - endpoints
+        verbs:
+          - get
+          - patch
+          - update
+          - create
+          - list
+          - watch
+          - delete
+      - apiGroups:
+          - ""
+        resources:
+          - pods
+        verbs:
+          - get
+          - list
+          - patch
+          - update
+          - watch
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    metadata:
+      labels:
+        app: ${APP_LABEL_NAME}
+        cluster-name: ${NAME}
+      name: ${NAME}
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: Role
+      name: ${NAME}
+    subjects:
+      - kind: ServiceAccount
+        name: ${NAME}
+parameters:
+  - name: NAME
+    description: |
+      The name of the application for labelling all artifacts.
+    displayName: Application Name
+    value: hni-patroni-postgresql
+  - name: PATRONI_SUPERUSER_USERNAME
+    description: |
+      Username of the superuser account for initialization.
+    displayName: Superuser Username
+    value: postgres
+  - name: PATRONI_SUPERUSER_PASSWORD
+    description: |
+      Password of the superuser account for initialization.
+    displayName: Superuser Password
+    generate: expression
+    from: "[a-z0-9]{32}"
+  - name: PATRONI_REPLICATION_USERNAME
+    description: |
+      Username of the replication account for initialization.
+    displayName: Replication Username
+    value: replication
+  - name: PATRONI_REPLICATION_PASSWORD
+    description: Password of the replication account for initialization.
+    displayName: Replication Password
+    generate: expression
+    from: "[a-z0-9]{32}"
+  - name: APP_DB_USERNAME
+    generate: expression
+    from: "[a-z0-9]{8}"
+  - name: APP_DB_NAME
+    value: registry
+  - name: APP_DB_PASSWORD
+    generate: expression
```

Comment:
- We were providing the passwords for database setup as part of jenkins job. In Patroni setup, we are auto generating the password. Which approach should we keep when Patroni set up is integrated in application setup Jenkins job?

Reply:
I think letting Patroni generate it might be fine? I don't think we'd really be removing and recreating these databases often.